<a href="https://colab.research.google.com/github/chemoinformatics-lecture/lecture-beginner/blob/main/lesson05_MOPAC/%E3%80%9003%E3%80%91mopac%E3%81%AE%E8%A8%88%E7%AE%97%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB%E3%81%AE%E7%A2%BA%E8%AA%8D_colab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# google driveへの接続

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# outファイルの読み取り

In [3]:
with open('/content/drive/MyDrive/data/mopac/phenol.arc', mode='r') as f:
    lines = f.readlines()
    for line in lines:
        print(line, end='')



                     SUMMARY OF  PM7 CALCULATION

                                                       MOPAC v22.0.4 Linux
                                                       Mon Sep 19 03:50:18 2022

           Empirical Formula: C6 H6 O  =    13 atoms

 PM7 EF
 c1ccc(cc1)O
 InChI=1S/C6H6O/c7-6-4-2-1-3-5-6/h1-5,7H


     GEOMETRY OPTIMISED USING EIGENVECTOR FOLLOWING (EF).     
     SCF FIELD WAS ACHIEVED                                   

          HEAT OF FORMATION       =        -22.15794 KCAL/MOL =     -92.70883 KJ/MOL
          GRADIENT NORM           =          0.72507          =       0.20110 PER ATOM
          DIPOLE                  =          1.30756 DEBYE   POINT GROUP:  Cs  
          NO. OF FILLED LEVELS    =         18
          IONIZATION POTENTIAL    =          9.235839 EV
          HOMO LUMO ENERGIES (EV) =         -9.236  0.035
          MOLECULAR WEIGHT        =         94.1128
          COSMO AREA              =        129.57 SQUARE ANGSTROMS
          COSM

# arcファイルの読み取り

In [ ]:
with open('phenol.arc', mode='r') as f:
    lines = f.readlines()
    for line in lines:
        print(line, end='')



                     SUMMARY OF  PM7 CALCULATION

                                                       MOPAC v22.0.4 MacOS
                                                       Sat Sep 17 21:36:16 2022

           Empirical Formula: C6 H6 O  =    13 atoms

 PM7 EF
 c1ccc(cc1)O
 InChI=1S/C6H6O/c7-6-4-2-1-3-5-6/h1-5,7H


     GEOMETRY OPTIMISED USING EIGENVECTOR FOLLOWING (EF).     
     SCF FIELD WAS ACHIEVED                                   

          HEAT OF FORMATION       =        -22.15781 KCAL/MOL =     -92.70830 KJ/MOL
          GRADIENT NORM           =          0.99656          =       0.27640 PER ATOM
          DIPOLE                  =          1.30764 DEBYE   POINT GROUP:  Cs  
          NO. OF FILLED LEVELS    =         18
          IONIZATION POTENTIAL    =          9.235653 EV
          HOMO LUMO ENERGIES (EV) =         -9.236  0.035
          MOLECULAR WEIGHT        =         94.1128
          COSMO AREA              =        129.57 SQUARE ANGSTROMS
          COSM

それぞれの意味

HEAT OF FORMATION       =        -22.15781 KCAL/MOL =     -92.70830 KJ/MOL
GRADIENT NORM           =          0.99656          =       0.27640 PER ATOM
DIPOLE                  =          1.30764 DEBYE   POINT GROUP:  Cs  
NO. OF FILLED LEVELS    =         18
IONIZATION POTENTIAL    =          9.235653 EV
HOMO LUMO ENERGIES (EV) =         -9.236  0.035
MOLECULAR WEIGHT        =         94.1128
COSMO AREA              =        129.57 SQUARE ANGSTROMS
COSMO VOLUME            =        119.27 CUBIC ANGSTROMS

# ファイルの読み取り方


In [12]:
# 使用するライブラリーの読み込み
import os
import glob
import pandas as pd

## 1つのarcファイルの読み取り

In [11]:
file_path = "/content/drive/MyDrive/data/mopac/phenol.arc"
name = os.path.splitext(os.path.basename(file_path))[0]
print(name)

with open(file_path, mode='r') as f:
    lines = f.readlines()
    #stripメソッドで空白や特定文字の削除
    lines_strip = [line.strip() for line in lines]
    #特定の文字列があるかを判定
    HOMO_line = [line for line in lines_strip if 'HOMO' in line]
    if HOMO_line != []:
      HOMO=float(str(HOMO_line[0][30:39]))
      print("HOMO: " + str(HOMO))

phenol
HOMO: -9.23


## 複数のarcファイルの読み取り